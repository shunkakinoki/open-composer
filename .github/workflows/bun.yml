name: Bun
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  workflow_dispatch:
concurrency:
  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.head_ref || github.sha }}
  cancel-in-progress: true
jobs:
  bun-build:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Build
        run: bun run build
  bun-lint:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Check
        run: bun run check
      - name: Format
        run: bun run format
      - name: Lint
        run: bun run lint
      - name: Type Check
        run: bun run type-check
  bun-test:
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - name: Setup Bun
        uses: shunkakinoki/actions/.github/actions/setup-bun@ba7baa1682a0248f78110dff5638e9a70b7703b2
      - name: Test
        run: bun run test
  bun-check:
    if: always()
    needs:
      - bun-build
      - bun-lint
      - bun-test
    runs-on: ubuntu-latest
    timeout-minutes: 3
    steps:
      - name: Alls Green
        uses: re-actors/alls-green@release/v1
        with:
          jobs: ${{ toJSON(needs) }}
