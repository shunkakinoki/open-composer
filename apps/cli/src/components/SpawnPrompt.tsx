import { Box, Text, useApp, useInput } from "ink";
import type React from "react";
import { useState } from "react";
import { AVAILABLE_AGENTS } from "../commands/spawn-command.js";

type Step = "session-name" | "agents" | "base-branch" | "create-pr" | "confirm";

interface SpawnPromptProps {
  onComplete: (config: SpawnConfig) => void;
  onCancel: () => void;
}

export interface SpawnConfig {
  sessionName: string;
  agents: string[];
  baseBranch: string;
  createPR: boolean;
}

export const SpawnPrompt: React.FC<SpawnPromptProps> = ({
  onComplete,
  onCancel,
}) => {
  const [step, setStep] = useState<Step>("session-name");
  const [sessionName, setSessionName] = useState("");
  const [selectedAgents, setSelectedAgents] = useState<string[]>([]);
  const [baseBranch, setBaseBranch] = useState("main");
  const [createPR, setCreatePR] = useState(false);
  const [currentAgentIndex, setCurrentAgentIndex] = useState(0);
  const { exit } = useApp();

  const toggleAgent = (agent: string) => {
    setSelectedAgents((prev) =>
      prev.includes(agent) ? prev.filter((a) => a !== agent) : [...prev, agent],
    );
  };

  const handleComplete = () => {
    const finalSessionName = sessionName.trim() || `spawn-${Date.now()}`;

    const config: SpawnConfig = {
      sessionName: finalSessionName,
      agents: selectedAgents.length > 0 ? selectedAgents : ["codex"], // default to codex if none selected
      baseBranch: baseBranch.trim() || "main",
      createPR,
    };

    onComplete(config);
    exit();
  };

  useInput(
    (input, key) => {
      if (key.escape || (key.ctrl && input === "c")) {
        onCancel();
        exit();
        return;
      }

      switch (step) {
        case "session-name":
          if (key.return) {
            setStep("agents");
          } else if (key.backspace || key.delete) {
            setSessionName(sessionName.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setSessionName(sessionName + input);
          }
          break;

        case "agents":
          if (input === " ") {
            const agent = AVAILABLE_AGENTS[currentAgentIndex];
            toggleAgent(agent);
          } else if (key.upArrow) {
            setCurrentAgentIndex((prev) =>
              prev > 0 ? prev - 1 : AVAILABLE_AGENTS.length - 1,
            );
          } else if (key.downArrow) {
            setCurrentAgentIndex((prev) =>
              prev < AVAILABLE_AGENTS.length - 1 ? prev + 1 : 0,
            );
          } else if (key.return) {
            setStep("base-branch");
          }
          break;

        case "base-branch":
          if (key.return) {
            if (baseBranch.trim()) {
              setStep("create-pr");
            }
          } else if (key.backspace || key.delete) {
            setBaseBranch(baseBranch.slice(0, -1));
          } else if (input && !key.ctrl && !key.meta) {
            setBaseBranch(baseBranch + input);
          }
          break;

        case "create-pr":
          if (
            key.leftArrow ||
            key.rightArrow ||
            input.toLowerCase() === "y" ||
            input.toLowerCase() === "n"
          ) {
            setCreatePR(!createPR);
          } else if (key.return) {
            setStep("confirm");
          }
          break;

        case "confirm":
          if (key.return) {
            handleComplete();
          }
          break;
      }
    },
    { isActive: true },
  );

  const renderStep = () => {
    switch (step) {
      case "session-name":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              ü™Ñ Spawn Interactive Session
            </Text>
            <Box marginTop={1}>
              <Text>name for the session: </Text>
              <Text color="green">{sessionName}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">
                Press Enter to continue (auto-generated if empty), Esc to cancel
              </Text>
            </Box>
          </Box>
        );

      case "agents":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              ü§ñ Select Agents
            </Text>
            <Box marginTop={1}>
              <Text>
                Select one or multiple coding agents: (use space to select,
                enter to confirm)
              </Text>
            </Box>
            <Box marginTop={1} flexDirection="column">
              {AVAILABLE_AGENTS.map((agent, index) => (
                <Box key={agent}>
                  <Text
                    color={selectedAgents.includes(agent) ? "green" : "gray"}
                  >
                    {selectedAgents.includes(agent) ? "[x]" : "[ ]"} {agent}
                    {index === currentAgentIndex && (
                      <Text color="yellow"> ‚Üê</Text>
                    )}
                  </Text>
                </Box>
              ))}
            </Box>
            <Box marginTop={1}>
              <Text color="gray">
                Use ‚Üë‚Üì to navigate, Space to toggle, Enter to confirm, Esc to
                cancel
              </Text>
            </Box>
          </Box>
        );

      case "base-branch":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              üåø Base Branch
            </Text>
            <Box marginTop={1}>
              <Text>select the branch to branch from: </Text>
              <Text color="green">{baseBranch}</Text>
              <Text color="gray">_</Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">
                Press Enter to continue (default: main), Esc to cancel
              </Text>
            </Box>
          </Box>
        );

      case "create-pr":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              üîó Create PRs
            </Text>
            <Box marginTop={1}>
              <Text>Create PRs for spawned worktrees?</Text>
            </Box>
            <Box marginTop={1}>
              <Text color={createPR ? "green" : "red"}>
                {createPR ? "[x] Yes" : "[ ] No"}
              </Text>
            </Box>
            <Box marginTop={1}>
              <Text color="gray">
                Use ‚Üê‚Üí or Y/N to toggle, Enter to confirm, Esc to cancel
              </Text>
            </Box>
          </Box>
        );

      case "confirm":
        return (
          <Box flexDirection="column">
            <Text bold color="cyan">
              ‚úÖ Confirm Spawn Configuration
            </Text>
            <Box marginTop={1}>
              <Text>
                Session Name:{" "}
                <Text color="green">
                  {sessionName.trim() || `spawn-${Date.now()}`}
                </Text>
              </Text>
            </Box>
            <Box marginTop={1}>
              <Text>
                Agents:{" "}
                <Text color="yellow">
                  {selectedAgents.length > 0
                    ? selectedAgents.join(", ")
                    : "codex"}
                </Text>
              </Text>
            </Box>
            <Box marginTop={1}>
              <Text>
                Base Branch:{" "}
                <Text color="yellow">{baseBranch.trim() || "main"}</Text>
              </Text>
            </Box>
            <Box marginTop={1}>
              <Text>
                Create PRs:{" "}
                <Text color={createPR ? "green" : "red"}>
                  {createPR ? "Yes" : "No"}
                </Text>
              </Text>
            </Box>
            <Box marginTop={2}>
              <Text color="gray">Press Enter to spawn, Esc to cancel</Text>
            </Box>
          </Box>
        );
    }
  };

  return (
    <Box flexDirection="column" padding={2}>
      {renderStep()}
    </Box>
  );
};
