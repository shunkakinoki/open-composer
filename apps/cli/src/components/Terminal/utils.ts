/**
 * Terminal utility functions
 * Inspired by termact: https://github.com/MasterGordon/termact/blob/main/src/utils.ts
 */

import type { TerminalStyle } from "./types.js";
import {
  bg,
  blink,
  bold,
  doubleUnderline,
  fg,
  hidden,
  inverse,
  italic,
  strikethrough,
  underline,
} from "./ansi.js";

export const applyStyle = (style: TerminalStyle): string => {
  let result = "";
  if (style.fg) result += fg(style.fg);
  if (style.bg) result += bg(style.bg);
  if (style.bold) result += bold();
  if (style.italic) result += italic();
  if (style.underline) result += underline();
  if (style.blink) result += blink();
  if (style.reverse) result += inverse();
  if (style.hidden) result += hidden();
  if (style.strikethrough) result += strikethrough();
  if (style.doubleUnderline) result += doubleUnderline();
  return result;
};

export const wrapText = (text: string, width: number): string[] => {
  const lines: string[] = [];
  const words = text.split(" ");
  let currentLine = "";

  for (const word of words) {
    if (currentLine.length + word.length + 1 <= width) {
      currentLine += (currentLine ? " " : "") + word;
    } else {
      if (currentLine) lines.push(currentLine);
      currentLine = word;
    }
  }

  if (currentLine) lines.push(currentLine);
  return lines;
};

export const truncateText = (text: string, maxLength: number): string => {
  if (text.length <= maxLength) {
    return text;
  }
  return text.slice(0, maxLength - 3) + "...";
};

export const padText = (
  text: string,
  width: number,
  align: "left" | "center" | "right" = "left",
): string => {
  if (text.length >= width) {
    return text.slice(0, width);
  }

  const padding = width - text.length;

  switch (align) {
    case "center": {
      const leftPad = Math.floor(padding / 2);
      const rightPad = padding - leftPad;
      return " ".repeat(leftPad) + text + " ".repeat(rightPad);
    }
    case "right":
      return " ".repeat(padding) + text;
    default:
      return text + " ".repeat(padding);
  }
};

export const getNtscGrayscale = (r: number, g: number, b: number): number => {
  return r * 0.299 + g * 0.587 + b * 0.114;
};
