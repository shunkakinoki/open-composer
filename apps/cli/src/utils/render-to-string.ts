import { createTestRenderer } from "@opentui/core/testing";
import type { ReactElement } from "react";

/**
 * Renders a React component to a string using OpenTUI
 * Useful for rendering UI components in Effect CLI commands
 *
 * Note: This is a simplified version that uses OpenTUI's test renderer
 * For full interactive rendering, use the main render function from @opentui/react
 */
export async function renderToString(element: ReactElement): Promise<string> {
  const { renderer, renderOnce, captureCharFrame } = await createTestRenderer({
    width: 100,
    height: 30,
  });

  // Note: We can't easily use the high-level render() here because it's designed
  // for full terminal interaction. For now, this returns an empty string.
  // Components should use direct terminal output if needed in CLI commands.

  // Clean up
  renderer.destroy();

  return "";
}
