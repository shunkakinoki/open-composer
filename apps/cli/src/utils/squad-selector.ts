import inquirer from "inquirer";
import type { LLMTask } from "@open-composer/agent-registry";
import { getAgentRegistry, getSquadLauncher } from "@open-composer/agent-registry";

/**
 * Start interactive squad selector
 */
export async function startSquadSelector(): Promise<void> {
  const registry = getAgentRegistry();
  const launcher = getSquadLauncher();

  console.log("\n🎮 Pokemon-Style Squad Manager\n");

  const { action } = await inquirer.prompt([
    {
      type: "list",
      name: "action",
      message: "What would you like to do?",
      choices: [
        { name: "📋 List all agents", value: "list" },
        { name: "✨ Create new squad", value: "create" },
        { name: "🚀 Quick squad (by task)", value: "quick" },
        { name: "📊 View statistics", value: "stats" },
        { name: "🚪 Exit", value: "exit" },
      ],
    },
  ]);

  switch (action) {
    case "list": {
      const agents = registry.getAllAgents();
      console.log(`\n📦 Found ${agents.length} agents:\n`);
      for (const agent of agents) {
        console.log(
          `  ${agent.pokemonAttributes.sprite} ${agent.name} (${agent.tier}) - ${agent.provider}`,
        );
      }
      break;
    }

    case "create": {
      console.log("\n🛠️  Create Squad - Not implemented in CLI yet");
      console.log("   Use the interactive UI components for full squad creation");
      break;
    }

    case "quick": {
      const { task } = await inquirer.prompt([
        {
          type: "list",
          name: "task",
          message: "Select task specialization:",
          choices: [
            { name: "🧭 Planner", value: "planner" },
            { name: "⚡ Programmer", value: "programmer" },
            { name: "🔍 Reviewer", value: "reviewer" },
            { name: "🌊 Router", value: "router" },
            { name: "🌿 Summarizer", value: "summarizer" },
          ],
        },
      ]);

      const team = launcher.createQuickSquad(task as LLMTask);
      console.log(`\n✅ Created squad: ${team.name}`);
      console.log(`   Agents: ${team.agents.map((a) => a.name).join(", ")}`);
      break;
    }

    case "stats": {
      const stats = registry.getStats();
      console.log("\n📊 Registry Statistics:");
      console.log(`   Total Agents: ${stats.totalAgents}`);
      console.log(`   Total Squads: ${stats.totalTeams}`);
      console.log(`   Starter: ${stats.tierCounts.starter || 0}`);
      console.log(`   Evolved: ${stats.tierCounts.evolved || 0}`);
      console.log(`   Legendary: ${stats.tierCounts.legendary || 0}`);
      break;
    }

    case "exit":
      console.log("\n👋 Goodbye!\n");
      return;
  }

  // Loop back to main menu
  await startSquadSelector();
}
