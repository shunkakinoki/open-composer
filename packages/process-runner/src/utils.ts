import * as Effect from "effect/Effect";
import {
  MAX_COMMAND_LENGTH,
  MAX_RUN_NAME_LENGTH,
  RUN_NAME_PATTERN,
} from "./constants.js";
import { ProcessRunnerError } from "./types.js";

// =============================================================================
// VALIDATION UTILITIES
// =============================================================================

export function validateRunName(
  runName: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!runName || typeof runName !== "string") {
    return Effect.fail(
      ProcessRunnerError("Run name must be a non-empty string"),
    );
  }

  if (runName.length > MAX_RUN_NAME_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Run name too long (max ${MAX_RUN_NAME_LENGTH} characters)`,
      ),
    );
  }

  if (!RUN_NAME_PATTERN.test(runName)) {
    return Effect.fail(
      ProcessRunnerError(
        "Run name can only contain letters, numbers, hyphens, and underscores",
      ),
    );
  }

  return Effect.succeed(runName);
}

export function validateCommand(
  command: string,
): Effect.Effect<string, ProcessRunnerError> {
  if (!command || typeof command !== "string") {
    return Effect.fail(
      ProcessRunnerError("Command must be a non-empty string"),
    );
  }

  if (command.length > MAX_COMMAND_LENGTH) {
    return Effect.fail(
      ProcessRunnerError(
        `Command too long (max ${MAX_COMMAND_LENGTH} characters)`,
      ),
    );
  }

  // Basic command injection prevention - only reject null bytes
  if (command.includes("\x00")) {
    return Effect.fail(
      ProcessRunnerError("Command contains invalid characters"),
    );
  }

  return Effect.succeed(command);
}

// =============================================================================
// UTILITY FUNCTIONS
// =============================================================================

export function withTimeout<T>(
  effect: Effect.Effect<T, ProcessRunnerError>,
  timeoutMs: number,
  timeoutMessage: string = "Operation timed out",
): Effect.Effect<T, ProcessRunnerError> {
  const timeoutEffect = Effect.async<T, ProcessRunnerError>((resume) => {
    const timeout = setTimeout(() => {
      resume(Effect.fail(ProcessRunnerError(timeoutMessage)));
    }, timeoutMs);
    return Effect.sync(() => clearTimeout(timeout));
  });

  return Effect.race(effect, timeoutEffect);
}
