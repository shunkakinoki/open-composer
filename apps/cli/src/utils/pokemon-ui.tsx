import type { AgentTeam, OpenComposerAgent } from "@open-composer/agent-registry";
import type { RegistryStats } from "@open-composer/analytics";
import { renderToString } from "./render-to-string.js";
import {
  AgentCard,
  AgentComparison,
  AgentList,
  EvolutionTree,
  RegistryStats as RegistryStatsComponent,
  TaskSpecializations,
  TeamDisplay,
} from "../components/PokemonUI.js";

/**
 * Pokemon-style UI renderer using React components
 * This is a utility wrapper that renders React components to strings for CLI output
 */
export class PokemonUI {
  /**
   * Render agent card
   */
  static renderAgentCard(agent: OpenComposerAgent): string {
    return renderToString(<AgentCard agent={agent} />);
  }

  /**
   * Render agent list
   */
  static renderAgentList(agents: OpenComposerAgent[]): string {
    return renderToString(<AgentList agents={agents} />);
  }

  /**
   * Render team display
   */
  static renderTeam(team: AgentTeam): string {
    return renderToString(<TeamDisplay team={team} />);
  }

  /**
   * Render evolution tree
   */
  static renderEvolutionTree(agent: OpenComposerAgent): string {
    return renderToString(<EvolutionTree agent={agent} />);
  }

  /**
   * Render task specializations
   */
  static renderTaskSpecializations(agent: OpenComposerAgent): string {
    return renderToString(<TaskSpecializations agent={agent} />);
  }

  /**
   * Render registry statistics
   */
  static renderStats(stats: RegistryStats): string {
    return renderToString(<RegistryStatsComponent stats={stats} />);
  }

  /**
   * Render agent comparison
   */
  static renderComparison(
    agent1: OpenComposerAgent,
    agent2: OpenComposerAgent,
  ): string {
    return renderToString(<AgentComparison agent1={agent1} agent2={agent2} />);
  }
}
