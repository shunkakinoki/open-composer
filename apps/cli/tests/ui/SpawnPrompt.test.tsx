import { describe, expect, mock, test } from "bun:test";

import { SpawnPrompt } from "../../src/components/SpawnPrompt.js";
import { render } from "../utils.js";

describe("SpawnPrompt", () => {
  const availableAgents = ["codex", "claude-code", "opencode"] as const;

  test("renders initial session name step correctly", async () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame, cleanup } = await render(
      <SpawnPrompt
        availableAgents={availableAgents}
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );
    expect(lastFrame()).toMatchSnapshot();
    cleanup();
  });

  test("renders with expected UI structure", async () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame, cleanup } = await render(
      <SpawnPrompt
        availableAgents={availableAgents}
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );

    const frame = lastFrame();
    // Check for main UI elements
    expect(frame).toContain("ðŸª„ Spawn Interactive Session");
    expect(frame).toContain("name for the session:");
    expect(frame).toContain("Press Enter to continue");
    expect(frame).toContain("Esc to cancel");
    cleanup();
  });

  test("component initializes without crashing", async () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    expect(async () => {
      await render(
        <SpawnPrompt
          availableAgents={availableAgents}
          onComplete={mockOnComplete}
          onCancel={mockOnCancel}
        />,
      );
    }).not.toThrow();
    cleanup();
  });

  test("accepts onComplete and onCancel callbacks", async () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame, cleanup } = await render(
      <SpawnPrompt
        availableAgents={availableAgents}
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );

    expect(lastFrame()).toBeDefined();
    expect(typeof mockOnComplete).toBe("function");
    expect(typeof mockOnCancel).toBe("function");
    cleanup();
  });

  test("displays session-related content initially", async () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame, cleanup } = await render(
      <SpawnPrompt
        availableAgents={availableAgents}
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );

    const frame = lastFrame();
    expect(frame).toContain("session");
    // Should not contain content from other steps
    expect(frame).not.toContain("ðŸ¤– Select Agents");
    expect(frame).not.toContain("ðŸŒ¿ Base Branch");
    cleanup();
  });

  test("has proper styling with emojis and colors", async () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame, cleanup } = await render(
      <SpawnPrompt
        availableAgents={availableAgents}
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );

    const frame = lastFrame();
    // Check for visual elements (emojis are preserved in the output)
    expect(frame).toContain("ðŸª„");
    cleanup();
  });

  test("provides user guidance text", async () => {
    const mockOnComplete = mock(() => {});
    const mockOnCancel = mock(() => {});

    const { lastFrame, cleanup } = await render(
      <SpawnPrompt
        availableAgents={availableAgents}
        onComplete={mockOnComplete}
        onCancel={mockOnCancel}
      />,
    );

    const frame = lastFrame();
    expect(frame).toContain("Press Enter");
    expect(frame).toContain("Esc");
    cleanup();
  });
});
