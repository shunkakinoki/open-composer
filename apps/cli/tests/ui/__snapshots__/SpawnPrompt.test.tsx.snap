// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`SpawnPrompt renders initial session name step correctly 1`] = `
"                                                                                
                                                                                
  ðŸª„ Spawn Interactive Session                                                  
                                                                                
  name for the session:                                                         
                                                                                
  _                                                                             
                                                                                
  Press Enter to continue (auto-generated if empty), Esc to cancel              
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
                                                                                
"
`;
